### PR TITLE
refactor: Adjust compose to fix changes breaking the TAF tests

### PR DESCRIPTION
**Pull request by lenny-intel**
This PR is dependent on https://github.com/edgexfoundry/edgex-go/pull/4308

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  TBD

## Testing Instructions
run `make docker` using branch from https://github.com/edgexfoundry/edgex-go/pull/4308
run make run ds-virtual dev from compose builder
Verify all service bootstrap successfully

### CODE REVIEW

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by bnevis-i — APPROVED** [review]

**Comment by lenny-intel** [comment]
recheck

**Review by marcpfuller — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Comment by cherrycl** [comment]
@lenny-intel The error due to the [edgex-go issue#4297](https://github.com/edgexfoundry/edgex-go/issues/4297).

**Comment by cloudxxx8** [comment]
recheck

**Comment by cloudxxx8** [comment]
recheck

**Comment by cherrycl** [comment]
Still blocked by [device-sdk-go issue #1293](https://github.com/edgexfoundry/device-sdk-go/issues/1293) .

**Review by jim-wang-intel — APPROVED** [review]

**Review by bnevis-i — COMMENTED** [review]
No objections.

**Review by marcpfuller — APPROVED** [review]

**Comment by lenny-intel** [comment]
@cherrycl , @cloudxxx8 , down to a single test, `NSMessagingGET005` failing.

**Comment by cherrycl** [comment]
This failure because the change of [edgex-go PR #4259](https://github.com/edgexfoundry/edgex-go/pull/4275), should update `MESSAGEBUS_INTERNAL` to `MESSAGEBUS` on compose file.

**Comment by lenny-intel** [comment]
> should update `MESSAGEBUS_INTERNAL` to `MESSAGEBUS` on compose file.

@cherrycl , added those updates to a commit on this PR.
https://github.com/edgexfoundry/edgex-compose/pull/328/commits/2bf8be4cf4eede3cda9c98402b9bff915f83cf0f

File: compose-builder/add-mqtt-messagebus.yml [hunk 1]
```diff
@@ -38,12 +38,10 @@ services:
...
-      MESSAGEBUS_INTERNAL_TYPE: mqtt
-      MESSAGEBUS_INTERNAL_PROTOCOL: tcp
-      MESSAGEBUS_INTERNAL_HOST: edgex-mqtt-broker
-      MESSAGEBUS_INTERNAL_PORT: 1883
-      MESSAGEBUS_INTERNAL_AUTHMODE: none
-      MESSAGEBUS_EXTERNAL_CLIENTID: core-command-ex
+      MESSAGEBUS_OPTIONAL_CLIENTID: core-command
+      MESSAGEBUS_AUTHMODE: none
+      MESSAGEBUS_HOST: edgex-mqtt-broker
+      EXTERNALMQTT_CLIENTID: core-command-ex
```
Comment (cherrycl) [inline compose-builder/add-mqtt-messagebus.yml:44]: Port and Type should be kept.

Comment (lenny-intel) [reply compose-builder/add-mqtt-messagebus.yml:44]: Yep, missed that the `mqtt-bus.env` file was not included on this service. Added it.

**Review by jim-wang-intel — APPROVED** [review]

**Pull request merged**